From 67b424626866d1a4b2e80c3e93698456042889fc Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Wed, 12 May 2010 10:55:37 +0100
Subject: [PATCH 105/105] gstreamer: Move some code inside pointer protection.

This can only happen if asserts are disabled, so this is somewhat
unlikely, but it's better to be safe.

This is unlikely to be related to the bug #59052, but in the event
that PA is running but both 'pulsesink' and 'fakesink' cannot be
created AND asserts are disabled, then this could cause some issues.

CCBUG: 59052
---
 gstreamer/devicemanager.cpp |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gstreamer/devicemanager.cpp b/gstreamer/devicemanager.cpp
index 6485094..3b0d4c3 100644
--- a/gstreamer/devicemanager.cpp
+++ b/gstreamer/devicemanager.cpp
@@ -392,10 +392,10 @@ void DeviceManager::updateDeviceList()
                 }
             }
         }
-    }
 
-    gst_element_set_state (audioSink, GST_STATE_NULL);
-    gst_object_unref (audioSink);
+        gst_element_set_state (audioSink, GST_STATE_NULL);
+        gst_object_unref (audioSink);
+    }
 }
 
 /**
-- 
1.7.1

